exporters/stats/signalfx/src/main/java/io/opencensus/exporter/stats/signalfx/SignalFxSessionAdaptor.java

```diff
@@ -17,172 +17,145 @@
+import com.google.common.collect.Lists;
 import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.DataPoint;
 import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.Datum;
 import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.Dimension;
 import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers.MetricType;
 import io.opencensus.common.Function;
-import io.opencensus.stats.Aggregation;
-import io.opencensus.stats.AggregationData;
-import io.opencensus.stats.AggregationData.CountData;
-import io.opencensus.stats.AggregationData.DistributionData;
-import io.opencensus.stats.AggregationData.LastValueDataDouble;
-import io.opencensus.stats.AggregationData.LastValueDataLong;
-import io.opencensus.stats.AggregationData.SumDataDouble;
-import io.opencensus.stats.AggregationData.SumDataLong;
-import io.opencensus.stats.View;
-import io.opencensus.stats.ViewData;
-import io.opencensus.tags.TagKey;
-import io.opencensus.tags.TagValue;
+import io.opencensus.common.Functions;
+import io.opencensus.metrics.LabelKey;
+import io.opencensus.metrics.LabelValue;
+import io.opencensus.metrics.export.Distribution;
+import io.opencensus.metrics.export.Metric;
+import io.opencensus.metrics.export.MetricDescriptor;
+import io.opencensus.metrics.export.MetricDescriptor.Type;
+import io.opencensus.metrics.export.Point;
+import io.opencensus.metrics.export.Summary;
+import io.opencensus.metrics.export.TimeSeries;
+import io.opencensus.metrics.export.Value;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
-import java.util.ListIterator;
-import java.util.Map;
 
-/*>>>
-import org.checkerframework.checker.nullness.qual.Nullable;
-*/
-
-/** Adapter for a {@code ViewData}'s contents into SignalFx datapoints. */
-@SuppressWarnings("deprecation")
+/** Adapter for a {@code Metric}'s contents into SignalFx datapoints. */
 final class SignalFxSessionAdaptor {
 
+  // Constant functions for Datum.
+  private static final Function<Double, Datum> datumDoubleFunction =
+      new Function<Double, Datum>() {
+        @Override
+        public Datum apply(Double arg) {
+          Datum.Builder builder = Datum.newBuilder();
+          builder.setDoubleValue(arg);
+          return builder.build();
+        }
+      };
+  private static final Function<Long, Datum> datumLongFunction =
+      new Function<Long, Datum>() {
+        @Override
+        public Datum apply(Long arg) {
+          Datum.Builder builder = Datum.newBuilder();
+          builder.setIntValue(arg);
+          return builder.build();
+        }
+      };
+  private static final Function<Distribution, Datum> datumDistributionFunction =
+      new Function<Distribution, Datum>() {
+        @Override
+        public Datum apply(Distribution arg) {
+          // Signal doesn't handle Distribution value.
+          // TODO(mayurkale): decide what to do with Distribution value.
+          throw new IllegalArgumentException("Distribution type are not supported");
```

Comment:
This will make the exporter crash. Consider logging this instead of throwing exceptions, since there will be users who have distribution metrics and export to multiple backends.